src/main/scala/com/fulcrumgenomics/vcf/MakeTwoSampleMixtureVcf.scala

```diff
@@ -0,0 +1,200 @@
+    |    'unknown_gt' filter applied
+    |  - If the tumor and the normal have more than one alternative allele between them the
+    |    'multi_allelic' filter will be applied
+    |  - In tumor/normal mode (as opposed to tumor-only) loci that have an alt allele in the normal
+    |    sample will have the 'alt_allele_in_normal' filter applied
+  """
+)
+class MakeTwoSampleMixtureVcf
+( @arg(flag="i", doc="Input VCF file.")  val input: PathToVcf,
+  @arg(flag="o", doc="Output VCF file.") val output: PathToVcf,
+  @arg(flag="t", doc="Name of the 'tumor' sample in the input VCF.")  val tumor: String,
+  @arg(flag="n", doc="Name of the 'normal' sample in the input VCF.") val normal: String,
+  @arg(flag="f", doc="What fraction of the mixture comes from the 'tumor' sample.") val tumorFraction: Double = 0.5,
+  @arg(flag="T", doc="Tumor only mode - only output tumor genotypes and don't filter sites.") val tumorOnly: Boolean = false,
+  @arg(flag="N", doc="Treat no-calls for either sample as hom-ref genotypes.") val noCallIsHomRef: Boolean = true,
+  @arg(flag="l", doc="Optional set of intervals to restrict to.") val intervals: Option[PathToIntervals] = None
+) extends FgBioTool {
+  if (tumorFraction < 0 || tumorFraction > 1) fail("Tumor fraction must be between 0 and 1.")
+  if (tumor == normal) fail("Tumor and Normal samples must be different samples.")
+
+  Io.assertReadable(input)
+  Io.assertCanWriteFile(output)
+  intervals.foreach(Io.assertCanWriteFile(_))
+
+  override def execute(): Unit = {
+    val in = new VCFFileReader(input.toFile, true)
+    if (!in.getFileHeader.getSampleNamesInOrder.contains(tumor))  fail(s"VCF does not contain tumor sample ${tumor}")
+    if (!in.getFileHeader.getSampleNamesInOrder.contains(normal)) fail(s"VCF does not contain normal sample ${normal}")
+
+    val out = if (tumorOnly) makeWriter(in.getFileHeader, "tumor") else makeWriter(in.getFileHeader, "tumor", "normal")
+    iterator(in, this.intervals).foreach { ctx =>
+      val oldTumorGt = genotypeOf(tumor, ctx)
+      val oldNormalGt = genotypeOf(normal, ctx)
+      val oldGts = Seq(oldTumorGt, oldNormalGt)
+
+      val alleles = (oldTumorGt.getAlleles ++ oldNormalGt.getAlleles ++ Some(ctx.getReference)).filterNot(_ == Allele.NO_CALL).toSet
+      val isMultiAllelic = alleles.size > 2
+      val isMonomorphic = alleles.size < 2
+
+      if (!isMonomorphic) {
+        val (newTumorGt, newNormalGt) = if (isMultiAllelic) {
+          (noCall(TumorSampleName), if (tumorOnly) None else Some(noCall(NormalSampleName)))
+        }
+        else {
+          val refAllele = ctx.getReference
+          val altAllele = ctx.getAlternateAllele(0)
+
+          // Build the Tumor genotype
+          val builder = new GenotypeBuilder(TumorSampleName)
+          val alleles = new util.ArrayList[Allele]()
+          if (oldGts.forall(_.isCalled)) {
+            if (oldGts.map(_.countAllele(refAllele)).sum + oldGts.count(_.isNoCall) > 0) alleles.add(refAllele)
```

Comment:
I am a bit confused by the `_.isNoCall` here given the command line option `noCallIsHomRef` isn't conditioned upon.  Are you thinking of a genotype with allele a No-call and the other a base?